File: src/server1.py
from flask import Flask, jsonify, request, make_response
import json

app = Flask(__name__)

def load_data():
    with open('db.json', 'r', encoding='utf-8') as f:
        return json.load(f)

def save_data(data):
    with open('db.json', 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=2, ensure_ascii=False)

def make_json_response(data, status=200):
    response = make_response(jsonify(data), status)
    response.headers['X-Server-ID'] = 'server1-python'
    return response


@app.route('/tickets', methods=['GET'])
def get_all_tickets():
    data = load_data()
    return make_json_response(data['tickets'])


@app.route('/tickets/<int:id>', methods=['GET'])
def get_ticket(id):
    data = load_data()
    for ticket in data['tickets']:
        if ticket['id'] == id:
            return make_json_response(ticket)
    return make_json_response({'error': 'Vstupenka nenalezena'}, 404)


@app.route('/tickets', methods=['POST'])
def create_ticket():
    data = load_data()
    new_ticket = request.get_json()

    if not new_ticket.get('name') or not new_ticket.get('table'):
        return make_json_response({'error': 'Chybí povinné údaje'}, 400)

    if data['tickets']:
        new_id = max(t['id'] for t in data['tickets']) + 1
    else:
        new_id = 1

    new_ticket['id'] = new_id
    data['tickets'].append(new_ticket)
    save_data(data)

    return make_json_response(new_ticket, 201)


@app.route('/tickets/<int:id>', methods=['PUT'])
def update_ticket(id):
    data = load_data()
    update_data = request.get_json()

    for i, ticket in enumerate(data['tickets']):
        if ticket['id'] == id:
            update_data['id'] = id
            data['tickets'][i] = update_data
            save_data(data)
            return make_json_response(update_data)

    return make_json_response({'error': 'Vstupenka nenalezena'}, 404)


if __name__ == '__main__':
    app.run(port=8001, debug=True)

File: lib/server1.js
const express = require('express');
const fs = require('fs');

const app = express();
app.use(express.json());

function loadData() {
    return JSON.parse(fs.readFileSync('db.json', 'utf-8'));
}

function saveData(data) {
    fs.writeFileSync('db.json', JSON.stringify(data, null, 2));
}

app.use((req, res, next) => {
    res.setHeader('X-Server-ID', 'server1-javascript');
    next();
});

app.get('/getDrinks/all', (req, res) => {
    const data = loadData();
    res.json(data.drinks);
});

app.get('/getDrink/byId/:id', (req, res) => {
    const data = loadData();
    const drink = data.drinks.find(d => d.id === parseInt(req.params.id));
    if (!drink) {
        return res.status(404).json({ error: 'Nápoj nenalezen' });
    }
    res.json(drink);
});

app.post('/createDrink', (req, res) => {
    const data = loadData();
    const newDrink = req.body;

    if (!newDrink.name || !newDrink.price) {
        return res.status(400).json({ error: 'Chybí povinné údaje' });
    }

    let newId = 1;
    if (data.drinks.length > 0) {
        newId = Math.max(...data.drinks.map(d => d.id)) + 1;
    }

    newDrink.id = newId;
    data.drinks.push(newDrink);
    saveData(data);

    res.status(201).json(newDrink);
});

app.put('/updateDrink/:id', (req, res) => {
    const data = loadData();
    const id = parseInt(req.params.id);
    const updateData = req.body;

    const index = data.drinks.findIndex(d => d.id === id);
    if (index === -1) {
        return res.status(404).json({ error: 'Nápoj nenalezen' });
    }

    updateData.id = id;
    data.drinks[index] = updateData;
    saveData(data);

    res.json(updateData);
});


app.listen(8001, () => {
    console.log('Server 1 běží na portu 8001');
});
